covid19-etl/etl/cityofchicago.py

```diff
@@ -0,0 +1,410 @@
+            "age_18_29": "18 to 29",
+            "age_30_39": "30 to 39",
+            "age_40_49": "40 to 49",
+            "age_50_59": "50 to 59",
+            "age_60_69": "60 to 69",
+            "age_70_79": "70 to 79",
+            "age_80_": "greater than 80",
+            "80_": "greater than 80",
+            "age_unknown": "unknown",
+            "unknown_age": "unknown",
+        }
+
+        # Race mapping for submitter id value from original dataset to value in Gen3 data dictionary
+        race_submitter_id = {
+            "latinx": "Hispanic or Latino",
+            "asian_non_latinx": "Asian",
+            "black_non_latinx": "Black or African-American",
+            "white_non_latinx": "White",
+            "other_non_latinx": "Other race",
+            "other_race_non_latinx": "Other race",
+            "unknown_race_eth": "Unknown",
+            "unknown_race_ethnicity": "Unknown",
+        }
+
+        # race mapping for value from original dataset to value in Gen3 data dictionary
+        race = {
+            "latinx": "Hispanic",
+            "asian_non_latinx": "Asian",
+            "black_non_latinx": "Black",
+            "white_non_latinx": "White",
+            "other_non_latinx": "Other",
+            "other_race_non_latinx": "Other",
+            "unknown_race_eth": "Unknown",
+            "unknown_race_ethnicity": "Unknown",
+        }
+
+        # Gender for value from original dataset to value in Gen3 data dictionary
+        gender = {
+            "female": "Female",
+            "male": "Male",
+            "unknown_gender": "Unknown",
+        }
+
+        record_type_mapping = {
+            "cases": "count",
+            "deaths": "deaths",
+            "hospitalizations": "hospitalizations",
+        }
+
+        record_type, submitter_value = header_value.split("_", maxsplit=1)
+        if submitter_value in age_group:
+            if "80" in submitter_value:
+                age_group_dict["AgeGroup"] = "80+"
+            else:
+                age_group_dict["AgeGroup"] = age_group[submitter_value].replace(
+                    " to ", "_"
+                )
+            summary_group_demographics_value_dict["age_group"] = age_group[
+                submitter_value
+            ]
+
+        elif submitter_value in race:
+            age_group_dict["Race"] = race_submitter_id[submitter_value]
+            summary_group_demographics_value_dict["race"] = race[submitter_value]
+            if submitter_value == "latinx":
+                summary_group_demographics_value_dict["ethnicity"] = "Hispanic"
+            else:
+                summary_group_demographics_value_dict["ethnicity"] = "Nonhispanic"
+
+        elif submitter_value in gender:
+            age_group_dict["Gender"] = gender[submitter_value]
+            if submitter_value == "unknown_gender":
+                summary_group_demographics_value_dict[
+                    "gender"
+                ] = "Unknown or Left Blank"
```

Comment:
why don't you use the `gender` mapping for this?

Reply:
for male and female both `submitter_id` and `gender` column value have same values according to the mapping but for "unknown_gender", the submitter id is suppose to have `unknown` and the `gender` column should have value as 
`Unknown or Left Blank`


|original value from dataset | submitter_id | gender |
|---------------------------|--------------|--------|
| male | male | male |
| female | female | female |
| unknown_gender | unknown | Unknown or Left Blank |


So I am using the mapping define to map from original value from dataset to the value in submitter id and because only unknown was different, I used an if condition. let me know what do you think about this

Reply:
makes sense